src/map/script.c

```diff
@@ -21585,6 +21585,34 @@ BUILDIN(useatcmd)
 	return true;
 }
 
+BUILDIN(checkpermission)
+{
+	struct map_session_data *sd;
+
+	if (script_hasdata(st, 3)) {
+		sd = map->id2sd(script_getnum(st, 3));
+	} else {
+		sd = script->rid2sd(st);
+	}
+
+	if (sd == NULL) {
+		script_pushint(st, 0);
+		return false;
+	}
+
+	if (script_isstringtype(st, 2)) {
+		// argument is an atcommand
+		const char *cmd = script_getstr(st, 2);
+		script_pushint(st, pc->can_use_command(sd, cmd));
+	} else if (script_isinttype(st, 2)) {
+		// argument is a permission const
+		enum e_pc_permission perm = script_getnum(st, 2);
+		script_pushint(st, pc_has_permission(sd, perm));
+	}
```

Comment:
just gonna remove the `if` part from the `else if` since the `v` in `BUILDIN_DEF(checkpermission, "v?")` ensures already that it's either string or integer